presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcDeletedRows.java

```diff
@@ -214,12 +217,20 @@ private void loadValidPositions()
         return deletedRows;
     }
 
-    private static Path createPath(String partitionLocation, AcidInfo.DeleteDeltaInfo deleteDeltaInfo, String fileName)
+    private static Path createPath(AcidInfo acidInfo, AcidInfo.DeleteDeltaInfo deleteDeltaInfo, String fileName)
     {
-        Path directory = new Path(partitionLocation, deleteDeltaSubdir(
+        Path directory = new Path(acidInfo.getPartitionLocation(), deleteDeltaSubdir(
                 deleteDeltaInfo.getMinWriteId(),
                 deleteDeltaInfo.getMaxWriteId(),
                 deleteDeltaInfo.getStatementId()));
+
+        if (acidInfo.isDirectInsert()) {
+            checkArgument(acidInfo.getBucketId().isPresent(), "bucketId is missing");
+            // When direct insert is enabled base and delta directories contain bucket_[id]_[attemptId] files
+            // but delete delta directories contain bucket files without attemptId.
+            return new Path(directory, format("bucket_%05d", acidInfo.getBucketId().getAsInt()));
```

Comment:
It should be just enough to cut the last `_.*` part of the original filename, right? It seems a bit safer as we are not recreating whole filename manually which may be error prone. E.g. here we are asssuming that bucket id is formated on 5 digits. while `BUCKET_WITH_OPTIONAL_ATTEMPT_ID_PATTERN` does not enforce that.

Also if we go with this approach we could get rid of bucketId from AcidInfo.

Would that work?

Reply:
Yep, that should work but if hive has direct insert then buckets are always 5 digits long. I can change that to strip attempt id from the string.

Reply:
I think I would prefer that approach. This way we may have less information in AcidUtils. If we get rid of bucketId from there it makes it easier to reason what is source of truth.

@findepi any though on this one?

Reply:
i think hive's logic is to recreate the filename. While it seems more err-prone, it may also be more future proof.